### DISCUSSION TITLE
Proposal: design change for the role view selector.

### DISCUSSION THREAD

**Discussion by duckbytes** [platelet-app/platelet#146] [Ideas]
Currently the dashboard shows you different controls and content depending on the role view you select:

![image](https://user-images.githubusercontent.com/32309223/233208456-f99a84ef-6476-41f4-a326-8b1d37e726fa.png)

This means that users need to select the correct role before starting their shift if they are both a coordinator and rider on the system. This selection persists in local storage.

I think a simpler design would be instead to make this a "filter". By default it could show all jobs that are assigned to that user as either coord or rider. A user is never going to be a coord and rider at the same time so I don't anticipate any confusion.

This means the user could start using the app straight away and will never be confused when jobs don't appear because they are using the incorrect role view.

Coords can also still have the "ALL" filter that shows all jobs on the system, but it must be enabled by them first. The selection will reset on page refresh to assigned as coord and rider jobs only.

![image](https://user-images.githubusercontent.com/32309223/233208327-23535299-c720-48a8-ba95-a19699ef81bc.png)

The drop down could instead show a multiple selection to apply or remove filters that you choose.

**Comment by Joetooker95** [comment]
Hi Theo,The current design seems more straight forwards, what is the reason for changing the design?Kind regardsJoe TookerOn 19 Apr 2023, at 23:02, Theo ***@***.***> wrote:﻿
Currently the dashboard shows you different controls and content depending on the role view you select:

This means that users need to select the correct role before starting their shift if they are both a coordinator and rider on the system. This selection persists in local storage.
I think a simpler design would be instead to make this a "filter". By default it could show all jobs that are assigned to that user as either coord or rider. A user is never going to be a coord and rider at the same time so I don't anticipate any confusion.
This means the user could start using the app straight away and will never be confused when jobs don't appear because they are using the incorrect role view.
Coords can also still have the "ALL" filter that shows all jobs on the system, but it must be enabled by them first. The selection will reset on page refresh to assigned as coord and rider jobs only.

The drop down could instead show a multiple selection to apply or remove filters that you choose.

**Reply by duckbytes** [reply]
The filter with multiple selections is a more common design pattern I see, and I think offers some flexibility because you can choose a combination of exactly what you want to see.

In the future for example, it could be used to show or hide upcoming scheduled jobs.

It also means for users that there is no extra step to take in 99% of cases when they log onto the app, because they'll immediately be able to see the information they need. Say if someone was previously coordinating a week ago and now had a riding shift, with the current design they would need to switch the dashboard to "RIDER" mode before they'll see any of their assigned jobs. With this design those jobs will be visible straight away.
